lib/analytics.js

```diff
@@ -259,17 +292,49 @@ Analytics.prototype.group = function(id, traits, options, fn) {
   if (is.object(id)) options = traits, traits = id, id = group.id();
   /* eslint-enable no-unused-expressions, no-sequences */
 
+  var plan = this.options.plan || {};
+  var traitPlans = plan.group || {};
+  var originalTraits = extend({}, traits);
+
+  for (var trait in traits) {
+    if (traits.hasOwnProperty(trait)) {
+      var traitPlan = traitPlans[trait];
+
+      if (traitPlan) {
+        this.log('group plan %o - %o', trait, traitPlan);
+        if (traitPlan.enabled === false) {
+          delete traits[trait];
+        }
+      } else if (traitPlans.__default) {
+        this.log('group default plan %o - %o', trait, traitPlans.__default);
+        if (!traitPlans.__default.enabled) {
+          delete traits[trait];
+        }
+      }
+    }
+  }
 
   // grab from group again to make sure we're taking from the source
   group.identify(id, traits);
 
+  // Send allowed traits to all integrations except Segment
   var msg = this.normalize({
     options: options,
     traits: group.traits(),
     groupId: group.id()
   });
+  extend(msg.integrations, { 'Segment.io': false });
```

Comment:
Do we still need this `'Segment.io': false`?

Reply:
I don't how the internals of a.js works but this seems safer and is future proof for when the segment integration is improved.